docs/Inspection-of-yum-Repositories.md

```diff
@@ -0,0 +1,125 @@
+# Inspection of yum Repositories
+
+When a Red Hat system is inspected we need to handle yum repositories. There are two aspects we have to consider:
+* Reading the repository data from the system
+* Storing the repository data in the manifest file
+
+## Reading the Repository Data from the System
+
+This section presents different ways of accessing the yum repository data and lists their advantages and disadvantages.
+
+### Parse output of `yum -v repolist all`
+* Advantages:
+  * Lists all repos
+* Disadvantages:
+  * Parsing not trivial (line breaks depend on length of output)
+  * Not all repo attributes accessible
+
+### Parse the repo files in `/etc/yum.repos.d`
+* Advantages:
+  * list (almost) all repo attributes
+* Disadvantages:
+  * Repos from plugins missing (e.g. suse manager's repos)
+
+### Accessing the python api
+* Advantages:
+  * All repos and repo attributes accessible (you need to know the attribute's name, though)
+* Disadvantages:
+  * It's not clear how to get the repo's url using the api, yet
+  * Copying script to target needed
+
+### Calling `yum-config-manager`
+* Advantages:
+  * Lists all repo attributes (incl. defaults)
+* Disadvantages:
+  * Hard to parse
+  * Disabled repos missing
+
```

Comment:
What about the variables used in repo files? Should we store them or the expanded value? Where do the values of the variables come from, and should we capture that as well?